src/gui/src/staGui.cpp

```diff
@@ -1352,4 +1363,168 @@ void TimingConeRenderer::annotateTiming(sta::Pin* source_pin)
+
+  QFormLayout* layout = new QFormLayout;
+
+  layout->addRow("Paths:", path_count_spin_box_);
+  layout->addRow("Command corner:", corner_box_);
+  layout->addRow("From:", from_line_);
+  layout->addRow("Through:", thru_line_);
+  layout->addRow("To:", to_line_);
+
+  connect(from_line_, SIGNAL(returnPressed()), this, SLOT(setFromPin()));
+  connect(thru_line_, SIGNAL(returnPressed()), this, SLOT(setThruPin()));
+  connect(to_line_, SIGNAL(returnPressed()), this, SLOT(setToPin()));
+
+  setLayout(layout);
+
+  const int min_characters = 25;
+  const auto font_metrics = from_line_->fontMetrics();
+  const int min_width = min_characters * font_metrics.averageCharWidth();
+  from_line_->setMinimumWidth(min_width);
+  thru_line_->setMinimumWidth(min_width);
+  to_line_->setMinimumWidth(min_width);
+
+  connect(corner_box_,
+          QOverload<int>::of(&QComboBox::currentIndexChanged),
+          [this](int index) {
+            if (index < 0 || index >= corner_box_->count()) {
+              return;
+            }
+            auto* corner = corner_box_->itemData(index).value<sta::Corner*>();
+            sta_->setCmdCorner(corner);
+          });
+}
+
+void TimingControlsDialog::populate()
+{
+  setPinSelections();
+
+  auto* current_corner = sta_->cmdCorner();
+  corner_box_->clear();
+  int selection = 0;
+  for (auto* corner : sta_->corners()->corners()) {
+    if (corner == current_corner) {
+      selection = corner_box_->count();
+    }
+    corner_box_->addItem(corner->name(), QVariant::fromValue(corner));
+  }
+
+  if (corner_box_->count() > 1) {
+    selection += 1;
+    corner_box_->insertItem(0, "All", QVariant::fromValue(static_cast<sta::Corner*>(nullptr)));
+    if (current_corner == nullptr) {
+      selection = 0;
+    }
+  }
+
+  corner_box_->setCurrentIndex(selection);
+}
+
+void TimingControlsDialog::setPinSelections()
+{
+  populateText(from_line_, from_);
+  populateText(thru_line_, thru_);
+  populateText(to_line_, to_);
+}
+
+void TimingControlsDialog::populateText(QLineEdit* line, const std::set<sta::Pin*>& pins)
+{
+  if (pins.empty()) {
+    line->clear();
+  } else {
+    auto* network = sta_->getDbNetwork();
+    QStringList text;
+    for (const auto* pin : pins) {
+      text.append(network->name(pin));
+    }
+    line->setText(text.join(" "));
+  }
+}
+
+void TimingControlsDialog::setFromPin()
+{
+  findPins(from_line_->text(), from_);
+  setPinSelections();
+}
+
+void TimingControlsDialog::setThruPin()
+{
+  findPins(thru_line_->text(), thru_);
+  setPinSelections();
+}
+
+void TimingControlsDialog::setToPin()
+{
+  findPins(to_line_->text(), to_);
+  setPinSelections();
+}
+
+void TimingControlsDialog::findPins(const QString& pin_names, std::set<sta::Pin*>& pins) const
+{
+  pins.clear();
+  if (pin_names.isEmpty()) {
+    return;
+  }
+  auto* network = sta_->getDbNetwork();
+
+  for (const QString pin_name : pin_names.split(" ")) {
+    sta::PatternMatch matcher(pin_name.toLatin1().constData());
```

Comment:
toLatin1 returns a temporary which is the source of my problems.

Reply:
fixed per your message